src/Repository/UserRepository.php

```diff
@@ -54,7 +70,18 @@ public function upgradePassword(PasswordAuthenticatedUserInterface $user, string
         }
 
         $user->setPassword($newHashedPassword);
-        $this->_em->persist($user);
         $this->_em->flush();
     }
+
+    public function compareTokensWhenChangingPassword(?string $tokenReset): ?User
+    {
+        $currentUser = $this->findOneBy(['tokenReset' => $tokenReset]);
+
+        if (null !== $currentUser) {
+            $message = 'User not found';
+            $this->logger->warning($message);
```

Comment:
You can return directly. The repository shouldn't log this. It will be logger on a higher level